sqlglot/dialects/snowflake.py

```diff
@@ -326,6 +327,21 @@ class Parser(parser.Parser):
             TokenType.SLASH,
         }
 
+        def _parse_lateral(self) -> t.Optional[exp.Lateral]:
+            lateral = super()._parse_lateral()
+            if not lateral:
+                return lateral
+
+            if isinstance(lateral.this, exp.Flatten):
+                table_alias = lateral.args.get("alias")
+                columns = [exp.to_identifier(col) for col in Snowflake.FLATTEN_COLUMNS]
+                if table_alias and not table_alias.args.get("columns"):
+                    table_alias.set("columns", columns)
+                elif not table_alias:
+                    lateral.set("alias", self.expression(exp.TableAlias, columns=columns))
```

Comment:
Don't we need to give an alias to the lateral source here? For example `_flattened` or something. We could also use the `exp.alias_` helper to set the alias / columns here.

Reply:
Didn't know if we could have a None alias. It looks like self._parse_table_alias _could_ have an empty alias since [it only tests](https://github.com/tobymao/sqlglot/blob/main/sqlglot/parser.py#L2229) for `not alias and not columns`. I can exp.alias_ here.

Reply:
It can have a `None` alias, but that'd look like the following query:

```sql
SELECT 1 FROM t AS (c)
```

Another nuance here is that if there are multiple `FLATTEN` calls without aliases we might need to create unique names for them by using something like `sqlglot.helper.find_new_name`, but we can address this later to if needed I guess.

Reply:
It seems like multiple lateral flatten without aliases wouldn't work anyway (hopefully 🤞)

Reply:
Ah I see, that's good. Snowflake seems to create the same name for flattens without aliases (i.e. `flatten`) and chokes due to duplicate aliases. Good catch 👍